From forest@alittletooquiet.net  Thu Jul  2 12:45:06 2009
From: Forest Bond <forest@alittletooquiet.net>
Date: Sat, 13 Jun 2009 07:38:47 -0400
Subject: Staging: vt6656: main_usb.c: Drop obsolete fsuid/fsgid accesses.
To: Greg KH <greg@kroah.com>
Cc: Larry Finger <Larry.Finger@lwfinger.net>
Message-ID: <20090613113847.GN26234@alittletooquiet.net>
Content-Disposition: inline


drivers/staging/vt6656/main_usb.c: Drop obsolete fsuid/fsgid accesses.

Signed-off-by: Forest Bond <forest@alittletooquiet.net>
Signed-off-by: Greg Kroah-Hartman <gregkh@suse.de>

---
 drivers/staging/vt6656/main_usb.c |   13 +++++++++----
 1 file changed, 9 insertions(+), 4 deletions(-)

--- a/drivers/staging/vt6656/main_usb.c
+++ b/drivers/staging/vt6656/main_usb.c
@@ -1655,15 +1655,17 @@ static UCHAR *Config_FileOperation(PSDev
     UCHAR    *buffer=NULL;
     struct file   *filp=NULL;
     mm_segment_t old_fs = get_fs();
-    int oldfsuid=0,oldfsgid=0;
+    //int oldfsuid=0,oldfsgid=0;
     int result=0;
 
     set_fs (KERNEL_DS);
-//Make sure a caller can read or write power as root
-   oldfsuid=current->fsuid;
-   oldfsgid=current->fsgid;
+    /* Can't do this anymore, so we rely on correct filesystem permissions:
+    //Make sure a caller can read or write power as root
+    oldfsuid=current->fsuid;
+    oldfsgid=current->fsgid;
     current->fsuid = 0;
     current->fsgid = 0;
+    */
 
     //open file
       filp = filp_open(config_path, O_RDWR, 0);
@@ -1697,8 +1699,11 @@ error1:
 
 error2:
   set_fs (old_fs);
+
+  /*
   current->fsuid=oldfsuid;
   current->fsgid=oldfsgid;
+  */
 
 if(result!=0) {
     if(buffer)
